src/Rules/Test.hs

```diff
@@ -63,13 +64,24 @@ testRules = do
         -- Execute the test target.
         buildWithCmdOptions env $ target (vanillaContext Stage2 compiler) RunTest [] []
 
+-- | Build extra programs required by testsuite
```

Comment:
The code below suggests that there are not only testsuite programs, but also testsuite libraries.

Reply:
@chitrak7 So, do you really need to handle libraries here?

Reply:
I have to handle rts library. I just cannot yet figure out how.

Reply:
Why do you need to handle `rts` here? How is it related to the testsuite?

Reply:
I still don't understand why `needTestsuiteBuilders` needs libraries. Can `testsuitePackages` contain library packages? Right now they only contain programs. If `testsuitePackages` can contain libraries, then you need to rename this function to something like `needTestsuitePackages` (instead of `Builders`).

Reply:
Some files are missing in rts library causing multiple failures. At the moment, I cannot figure out which files are missing. Or they may be missing includes. I have to look into this matter further.

Reply:
I don't understand how `rts` relates to this question. You don't have `rts` in `testsuitePackages`.

The comment for this function says "Build extra programs required by testsuite", but the implementation can also build libraries. This is an inconsistency that should be fixed.

Reply:
Resolved